### PR TITLE
w3m connect state improvments

### PR DESCRIPTION
**Pull request by kacperoak**
Migrate to getMobileListing to fetch wallets
Get a wallet screen
Move fetched wallet in to W3M state from ConnectYourWalletUI 
Move logic from ConnectYourScreen to modal initialization 

Test flow: :
1) Moved fetch wallet data logic:
		Run sample dapp -> press connect with selected chain -> wallet data should be already downloaded (e.g. icons of wallets in the carousel)
2) Adding the get a wallet screen:
Open web3modal -> press "I dont have a wallet" -> it should open state with a list of recommended wallets. You can additionally test redirect to playstore.

### CODE REVIEW

File: web3/modal/src/main/kotlin/com/walletconnect/web3/modal/data/explorer/ExplorerRepository.kt [hunk 1]
```diff
@@ -5,6 +5,7 @@ import com.walletconnect.android.internal.common.wcKoinApp
 import com.walletconnect.web3.modal.network.ExplorerService
 import com.walletconnect.web3.modal.network.model.ExplorerResponse
 
+private const val PLATFORM_ANDROID = "android"
```
Comment (Elyniss) [inline web3/modal/src/main/kotlin/com/walletconnect/web3/modal/data/explorer/ExplorerRepository.kt:8]: Nitpick: I would put it in companion object of ExplorerRepository

File: web3/modal/src/main/kotlin/com/walletconnect/web3/modal/domain/configuration/Config.kt [hunk 2]
```diff
@@ -10,9 +10,8 @@ sealed class Config {
 
     @JsonClass(generateAdapter = true)
     data class Connect(
```
Comment (Elyniss) [inline web3/modal/src/main/kotlin/com/walletconnect/web3/modal/domain/configuration/Config.kt:12]:
```
internal const val CONFIGURATION = "modal_config"
```

From above. Is this used somewhere?

**Review by Elyniss — APPROVED** [review]

**Comment by kacperoak** [comment]
changing this PR in the draft. We are waiting for the change of the endpoint in the explorer api from getMobileListing to getAndroidListings

File: web3/modal/src/main/kotlin/com/walletconnect/web3/modal/data/explorer/ExplorerRepository.kt (hunk 1 above)
Comment (kacperoak) [reply web3/modal/src/main/kotlin/com/walletconnect/web3/modal/data/explorer/ExplorerRepository.kt:8]: After api changes i will remove it

File: web3/modal/src/main/kotlin/com/walletconnect/web3/modal/domain/model/Wallet.kt [hunk 3]
```diff
@@ -1,21 +1,24 @@
...
     val name: String,
-    val images: WalletIcons,
+    val imageId: String,
     val nativeLink: String,
     val universalLink: String?,
+    // TODO Playstore link should be non null when `platforms` parameter will be fixed in the api
     val playStoreLink: String?,
-)
+) {
+    val imageUrl: String
```
[the remark this answers is not in the document]
Comment (kacperoak) [reply web3/modal/src/main/kotlin/com/walletconnect/web3/modal/domain/model/Wallet.kt:14]: Sure i will move the url into const

**Review by jakubuid — APPROVED** [review]

**Pull request merged**